src/eval/cast.rs

```diff
@@ -181,6 +182,30 @@ cast_to_value! {
     v: String => Value::Str(v.into())
 }
 
+cast_from_value! {
+    MathClass,
+    "operator" => MathClass::Large,
+    "opening" => MathClass::Opening,
+    "closing" => MathClass::Closing,
+    "binary" => MathClass::Binary,
+    "relation" => MathClass::Relation,
+    "normal" => MathClass::Normal,
+    "punctuation" => MathClass::Punctuation,
+}
+
+cast_to_value! {
+    v: MathClass => Value::from(match v {
+        MathClass::Large => "operator",
+        MathClass::Opening => "opening",
+        MathClass::Closing => "closing",
+        MathClass::Binary => "binary",
+        MathClass::Relation => "relation",
+        MathClass::Normal => "normal",
+        MathClass::Punctuation => "punctuation",
+        _ => "unused math class",
```

Comment:
This should either panic or just list all variants (even if they are not in the list above).